### DISCUSSION TITLE
"Cannot find model _Migration": Do I need to configure _Migration somewhere?

### DISCUSSION THREAD

**Discussion by rjmackay** [sensedeep/onetable-cli#13] [Q&A]
The docs aren't clear on all the things I need to get a working set up to start with.
Do you need to define a schema in migrate.json? Do I need to provide a schema for _Migration myself?

**Comment by mobsense** [comment]
No, you don't define a schema in the migrate.json.

Each migration will refer to a schema that applies for that migration version level.

I'll do another pass over the docs.

Here is our migration.json we use for SenseDeep. We use service-keys.json to provide the crypto setup and product.json to provide the onetable setup:

```
    "crypto": {
        "primary": {
            "cipher": "aes-256-gcm",
            "password": "password ....."
        }
    },
```

```
    "onetable": {
        "created": "created",
        "hidden": true,
        "isoDates": false,
        "metrics": true,
        "name": "sensedb",
        "nulls": false,
        "typeField": "type",
        "updated": "updated",
        "timestamps": true
    },

```

```
    config: ['../pak.json', '../product.json', '../CONFIG/service-keys.json'],
    profiles: {
        dev: {
            dir: './migrations',
            name: 'sensedb',
            endpoint: 'http://localhost:8000',
            region: 'localhost'
        },
        qa: {
            name: 'sensedb',
            arn: 'arn:aws:lambda:us-east-1:61852111111:function:migrate-qa-invoke'
        },
        prod: {
            name: 'sensedb',
            arn: 'arn:aws:lambda:us-east-1:505157811111:function:migrate-prod-invoke'
        }
    }
}

```

**Comment by mobsense** [comment]
I'd recommend you look at the [onetable-controller repo that ](https://github.com/sensedeep/onetable-controller) that has the full server side implemented for you.
